### PR TITLE
Saas12312/elasticsearch cron delete old logs

### PR DESCRIPTION
**Pull request by prem-fissionhq**
https://dimagi-dev.atlassian.net/browse/SAAS-12312
##### ENVIRONMENTS AFFECTED

Staging

Adding a Cron Template to delete es logs older than 7 days and not going to use Logrotate to manage elasticsearch logs.

### CODE REVIEW

File: src/commcare_cloud/ansible/roles/elasticsearch/defaults/main.yml [hunk 1]
```diff
@@ -17,3 +17,5 @@ backup_es_swift: no
 elasticsearch_enable_inline_groovy_scripts: True
 elasticsearch_service_name: elasticsearch
 elasticsearch_node_name: "{{ hostname }}"
+elasticsearch_deletelogs: False
+delete_older_logs_days: 7
```
Comment (dannyroberts) [inline src/commcare_cloud/ansible/roles/elasticsearch/defaults/main.yml:21]: All ansible variable names are global, so this has to start with elasticsearch to make it clear and not conflict. So maybe something like `elasticsearch_log_retention_days: 7`. Also I'd rename `elasticsearch_deletelogs` to `elasticsearch_delete_old_logs`

File: src/commcare_cloud/ansible/roles/elasticsearch/tasks/main.yml [hunk 2]
```diff
@@ -231,3 +231,28 @@
     mode: 0640
   notify: reload monit
   tags: monit
+
+- name: Copy elasticsearch log_retention file
+  template:
+    src: "delete_older_eslogs.sh.j2"
```
Comment (dannyroberts) [inline src/commcare_cloud/ansible/roles/elasticsearch/tasks/main.yml:237]: nit: can we change `delete_older_eslogs` to `delete_older_es_logs`?

**Review by dannyroberts — COMMENTED** [review]
Looks good. Comments around naming

**Comment by prem-fissionhq** [comment]
Corrected Danny, Please review. 
Thanks

File: src/commcare_cloud/ansible/roles/elasticsearch/tasks/main.yml [hunk 3]
```diff
@@ -231,3 +231,28 @@
...
+
+- name: Create Daily Cron job to delete elasticsearch_logs
+  cron:
+    name: "delete older elasticsearch logs"
+    job: "/usr/local/sbin/delete_older_es_logs.sh {{ elasticsearch_log_retention_days }}"
+    minute: 20
+    hour: "{{ nadir_hour|default(0) }}"
+    weekday: "1,2,3,4,5,6"
+    user: root
+    cron_file: delete_eslogs
```
Comment (dannyroberts) [inline src/commcare_cloud/ansible/roles/elasticsearch/tasks/main.yml:255]: nit: sorry I should have pointed this one out to. Consider changing to the more consistent/specific `delete_older_es_logs`

Comment (prem-fissionhq) [reply src/commcare_cloud/ansible/roles/elasticsearch/tasks/main.yml:255]: @dannyroberts I noticed it and changed yesterday to `delete_es_logs`, will modify it to `delete_older_es_logs`

```
(staging) ansible@es6-staging:/etc/cron.d$ ls -lrt
total 20
-rw-r--r-- 1 root root 396 Dec 23  2017 sysstat
-rw-r--r-- 1 root root 589 Jan 14  2020 mdadm
-rw-r--r-- 1 root root 190 Jun  5  2020 popularity-contest
-rw-r--r-- 1 root root 563 May 17 14:28 awsagent-update
-rw-r--r-- 1 root root 110 Jun 24 06:22 delete_es_logs
```

Comment (dannyroberts) [reply src/commcare_cloud/ansible/roles/elasticsearch/tasks/main.yml]: Great thanks

**Review by dannyroberts — APPROVED** [review]

**Comment by prem-fissionhq** [comment]
Applied on India env.

**Pull request merged**